rpcs3/Emu/Cell/PPUThread.cpp

```diff
@@ -2230,20 +2372,69 @@ extern void ppu_initialize()
 		prx_list.emplace_back(&prx);
 	});
 
+	// If empty we have no indication for cache state, check everything
+	bool compile_fw = prx_list.empty();
+
+	// Check preloaded libraries cache
+	for (auto ptr : prx_list)
+	{
+		compile_fw |= ppu_initialize(*ptr, true);
+	}
+
+	const std::string firmware_sprx_path = vfs::get("/dev_flash/sys/external/");
+
+	std::vector<std::string> dir_queue;
+		
+	if (compile_fw)
+	{
+		dir_queue.emplace_back(firmware_sprx_path);
+	}
+
+	// Avoid compilation if main's cache exists or it is a standalone SELF with no PARAM.SFO
+	if (compile_main && !Emu.GetTitleID().empty())
```

Comment:
Seems like a good choice 👍. Now users cannot compile main without having compiled the rest.